pkg/services/sso/keycloak_service.go

```diff
@@ -213,12 +214,12 @@ func (kc *masService) CreateServiceAccount(accessToken string, serviceAccountReq
 	orgId := auth.GetOrgIdFromClaims(claims)
 	ownerAccountId := auth.GetAccountIdFromClaims(claims)
 	owner := auth.GetUsernameFromClaims(claims)
-	isAllowed, err := kc.checkAllowedServiceAccountsLimits(accessToken, kc.GetConfig().MaxAllowedServiceAccounts, ownerAccountId)
+	isAllowed, err := kc.checkAllowedServiceAccountsLimits(accessToken, kc.GetConfig().MaxAllowedServiceAccounts, orgId)
 	if err != nil { //5xx
 		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to create service account")
 	}
 	if !isAllowed { //4xx over requesters' limit
-		return nil, errors.Forbidden("Max allowed number:%d of service accounts for user:%s has reached", kc.GetConfig().MaxAllowedServiceAccounts, owner)
+		return nil, errors.NewWithCause(errors.ErrorMaxLimitForServiceAccountsReached, err, "Max allowed number:%d of service accounts for user in org:%s has reached", kc.GetConfig().MaxAllowedServiceAccounts, orgId)
```

Comment:
Here `err` is always `nil`.
Change to:
```suggestion
		return nil, errors.MaxLimitForServiceAccountReached("Max allowed number:%d of service accounts for user in org:%s has reached", kc.GetConfig().MaxAllowedServiceAccounts, orgId)
```

Reply:
make sense. added